js/searchbar/customBangs.js

```diff
@@ -12,8 +12,126 @@ var places = require('places/places.js')
 var contentBlockingToggle = require('navbar/contentBlockingToggle.js')
 var taskOverlay = require('taskOverlay/taskOverlay.js')
 var bookmarkConverter = require('bookmarkConverter.js')
+var searchbarPlugins = require('searchbar/searchbarPlugins.js')
+var tabEditor = require('navbar/tabEditor.js')
+var formatRelativeDate = require('util/relativeDate.js')
+
+function moveToTask(text) {
+  /* disabled in focus mode */
+  if (focusMode.enabled()) {
+    focusMode.warn()
+    return
+  }
+
+  // remove the tab from the current task
+
+  var currentTab = tabs.get(tabs.getSelected())
+  tabs.destroy(currentTab.id)
+
+  // make sure the task has at least one tab in it
+  if (tabs.count() === 0) {
+    tabs.add()
+  }
+
+  var newTask = getTaskByNameOrNumber(text.toLowerCase())
+
+  if (newTask) {
+    newTask.tabs.add(currentTab, { atEnd: true })
+  } else {
+    // create a new task with the given name
+    newTask = tasks.get(tasks.add(undefined, tasks.getIndex(tasks.getSelected().id) + 1))
+    newTask.name = text
+
+    newTask.tabs.add(currentTab)
+  }
+
+  browserUI.switchToTask(newTask.id)
+  browserUI.switchToTab(currentTab.id)
+
+
+  taskOverlay.show()
+
+  setTimeout(function () {
+    taskOverlay.hide()
+  }, 600)
+}
+
+function switchToTask(text) {
+  /* disabled in focus mode */
+  if (focusMode.enabled()) {
+    focusMode.warn()
+    return
+  }
+
+  text = text.toLowerCase()
+
+  // no task was specified, show all of the tasks
+  if (!text) {
+    taskOverlay.show()
+    return
+  }
+
+  var task = getTaskByNameOrNumber(text)
+
+  if (task) {
+    browserUI.switchToTask(task.id)
+  }
+}
+
+// returns a task with the same name or index ("1" returns the first task, etc.)
+// In future PR move this to task.js
+function getTaskByNameOrNumber(text) {
+  const textAsNumber = parseInt(text)
+
+  return tasks.find((task, index) => (task.name && task.name.toLowerCase() === text) || index + 1 === textAsNumber
+  )
+}
+
+// return an array of dicts, sorted by last task activity
+// if a search string is present (and not a number) filter the results with a basic fuzzy search
+function searchAndSortTasks(text) {
+  var sortLastActivity = tasks.map(t => Object.assign({}, { task: t }, { lastActivity: tasks.getLastActivity(t.id) }))
+
+  sortLastActivity = sortLastActivity.sort(function (a, b) {
+    return b.lastActivity - a.lastActivity
+  })
+
+  var isSingleNumber = /^\d+$/.test(text)
+
+  if (text !== '' ? !isSingleNumber : !isSingleNumber) { //lXOR
```

Comment:
What's the reason for disallowing fuzzy search for names that are numbers? You could have task names that are long numbers, and it seems useful to support fuzzy search in that case.

(Also, this can be simplified to `!isSingleNumber` I think?)

Reply:
This is trying to remain consistent with previous functionality, e.g. typing in 1 (and only that) should move to task 1. 

L101 is a logical XOR (I'm pretty sure). SO the idea is if the bang has no text OR is only numbers draw the list sorted by just recent activity. Now you bring it up, I suppose if the task is JUST a number, it should probably display the list in task index order (rather than recent activity).

However tasks with numbers in the name e.g. My Bean Recipes 22/05 should still fuzzy search.

I agree it is confusing. I never switch task by index anyway....